patches/components-browser_ui-site_settings-android-java-src-org-chromium-components-browser_ui-site_settings-SiteSettingsCategory.java.patch

```diff
@@ -0,0 +1,44 @@
+diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SiteSettingsCategory.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SiteSettingsCategory.java
+index 5c0f84d697b29..0944d227e28c5 100644
+--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SiteSettingsCategory.java
++++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/SiteSettingsCategory.java
+@@ -42,7 +42,7 @@ public class SiteSettingsCategory {
+             Type.CLIPBOARD, Type.COOKIES, Type.IDLE_DETECTION, Type.DEVICE_LOCATION,
+             Type.JAVASCRIPT, Type.MICROPHONE, Type.NFC, Type.NOTIFICATIONS, Type.POPUPS,
+             Type.PROTECTED_MEDIA, Type.SENSORS, Type.SOUND, Type.USB, Type.VIRTUAL_REALITY,
+-            Type.USE_STORAGE})
++            Type.USE_STORAGE, Type.AUTOPLAY})
+     @Retention(RetentionPolicy.SOURCE)
+     public @interface Type {
+         // All updates here must also be reflected in {@link #preferenceKey(int)
+@@ -70,10 +70,11 @@ public class SiteSettingsCategory {
+         int BLUETOOTH = 20;
+         int VIRTUAL_REALITY = 21;
+         int USE_STORAGE = 22;
++        int AUTOPLAY = 23;
+         /**
+          * Number of handled categories used for calculating array sizes.
+          */
+-        int NUM_ENTRIES = 23;
++        int NUM_ENTRIES = 24;
+     }
+ 
+     private final BrowserContextHandle mBrowserContextHandle;
```

Comment:
I believe we have a way to replace static method calls with calls to a different method so can do BraveSiteSettingsCategory.contentSettingsType which calls SiteSettingsCategory.contentSettingsType and then checks for autoplay if the return is -1

Reply:
That's what I did with the rest of the java codes but it is impossible to do here because (1) the definition is annotated and to redirect the definition we need to do some very hacky bytecode manipulation (which I am not sure is possible at all), and if the descendant class has values not in the annotation compiler will probably complain too. (2) the value is extensive used in `site_settings` and some of them are not static, which means I need to do the hack of changing all type definitions in a function body with bytecode on all of them, which IMO is worse than just using a patch.